### PR TITLE
Create estimateZernikesBase class to prepare for corner wavefront sensor task.

### PR DESCRIPTION
**Pull request by jbkalmbach**
Since we will have two different estimateZernikes tasks (one for FAM and one for CWFS) we should abstract what we can into a base class to minimize code duplication.

### CODE REVIEW

**Comment by teweitsai** [comment]
Please update the following documents:
1. `doc/versionHistory.rst`
2. `doc/uml//taskClass.uml`
3. `doc/content.rst`
4. `ups/ts_wep.table` and `Jenkinsfile` for the package: `ctrl_mpexec`.

Thanks!

File: doc/content.rst [hunk 1]
```diff
@@ -145,6 +145,9 @@ This module has the tasks to run WEP as a pipeline with Gen 3 LSST DM middleware
 * **GenerateDonutCatalogOnlineTask**: Gen 3 middleware task to take pointing information and create a catalog of donut sources in that pointing.
 * **DonutStamp**: Storage class for a single donut postage stamp and associated metadata.
 * **DonutStamps**: Gen 3 Butler readable storage class for a list of DonutStamp objects with helper functions to get metadata and to save DonutStamps object as FITS file.
+* **EstimateZernikesBaseTaskConnections**: Base Connections class for EstimateZernikes tasks.
+* **EstimateZernikesBaseTaskConfig**: Base Configuration class for EstimateZernikes tasks.
+* **EstimateZernikesBaseTask**: Base Class for EstimateZernikes tasks.
```
Comment (teweitsai) [inline doc/content.rst]: "Base Class" --> "Base class".

But I am not sure for "Base Configuration" and "Base Connections". Or are you intentionally to use the capital word here (Configuration + Connections)?

File: python/lsst/ts/wep/task/EstimateZernikesBase.py [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+        dtype=int,
+        default=40,
+    )
+
+
+class EstimateZernikesBaseTask(pipeBase.PipelineTask):
+    """
+    Base class for Zernike estimation tasks.
+
+    Subclasses must implement _DefaultName
```
Comment (teweitsai) [inline python/lsst/ts/wep/task/EstimateZernikesBase.py]: Please add the period in the end of sentence.

File: tests/task/test_estimateZernikesBase.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        self.assertEqual(np.shape(zernCoeff), (len(donutStampsExtra), 19))
+
+    def testCombineZernikes(self):
+
+        testArr = np.zeros((2, 19))
+        testArr[1] += 2.0
+        np.testing.assert_array_equal(self.task.combineZernikes(testArr), np.ones(19))
+
+    @classmethod
+    def tearDownClass(cls):
```
Comment (teweitsai) [inline tests/task/test_estimateZernikesBase.py]: Please consider to reorder this `tearDownClass()` to be just after the `setUpClass()`. This is unrelated to the code itself but let future's developer can add the new test case from the final existed test case as straight-forward.

**Review by teweitsai — APPROVED** [review]
This update looks very good to me. The add of new base class helps the reuse of existed code. Thanks!

File: doc/content.rst (hunk 1 above)
Comment (jbkalmbach) [reply doc/content.rst]: No, these can all be lower case.

**Pull request merged**